src/lib/utils/plainTextFromTree.ts

```diff
@@ -0,0 +1,24 @@
+import React from "react"
+
+export function plainTextFromTree(root: React.ReactNode): string | any {
+  function traverse(node: React.ReactNode, currentProgress: string): string {
+    if (typeof node === "string") {
+      return node
+    }
+
+    if (Array.isArray(node)) {
+      let result = ""
+      for (const child of node) {
+        result += traverse(child, currentProgress)
+      }
+      return result
+    }
+
+    if (React.isValidElement(node)) {
+      const children = React.Children.toArray((node.props as any).children)
+      return traverse(children, currentProgress)
+    }
```

Comment:
it just occurred to me that we're not handling nulls, numbers, booleans, and other truthy objects properly. It should probably be:

- null/undefined/true/false - do not get rendered
- arrays - get traversed
- react elements - have their children traversed if present
- everything else - gets coerced to a string

Reply:
By "do not get rendered", would you expect it to throw an error? In this case it seems like it could just return `null` or `undefined`?

i.e.
```javascript
if (node === null || typeof node === "boolean" || typeof node === "undefined") {
  return null
}
```

Reply:
No it wouldn't throw an error, we want to replicate what react would render if you put the value, e.g. in an empty div. So for null/undefined/true/false it would be the empty string